indexing-service/src/main/java/io/druid/indexing/overlord/ThreadPoolTaskRunner.java

```diff
@@ -335,6 +379,11 @@ public TaskStatus call()
 
       try {
         log.info("Running task: %s", task.getId());
+        TaskRunnerUtils.notifyLocationChanged(
```

Comment:
There's a race here, right? where you have announced something before it is really running.

Does this also mean the overlord will be aware of the location change before the overlord even knows the task is actually running? or is the overlord's awareness of a successful start before here?

At the very least the plumber school hasn't started yet and having clarity on this order somewhere in the docs (or javaDocs) would be handy